some ui tests print traits differently on big-endian

version 1.44.1 [s390x](https://buildd.debian.org/status/fetch.php?pkg=rustc&arch=s390x&ver=1.44.1%2Bdfsg1-1&stamp=1595858692&raw=0) [ppc64](https://buildd.debian.org/status/fetch.php?pkg=rustc&arch=ppc64&ver=1.44.1%2Bdfsg1-1&stamp=1595858223&raw=0) [powerpc](https://buildd.debian.org/status/fetch.php?pkg=rustc&arch=powerpc&ver=1.44.1%2Bdfsg1-1&stamp=1595859982&raw=0)

I will be adding an `ignore-endian-big` feature as part of #74829, just wanted to confirm these failures are OK to ignore:

~~~~
---- [ui] ui/consts/const-eval/ub-enum.rs stdout ----
[ .. omitted for brevity .. ]
------------------------------------------
diff of stderr:

90	  --> $DIR/ub-enum.rs:90:1
91	   |
92	LL | const BAD_UNINHABITED_WITH_DATA1: Result<(i32, Never), (i32, !)> = unsafe { mem::transmute(1u64) };
-	   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed: encountered a value of the never type `!` at .<enum-variant(Err)>.0.1
+	   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed: encountered a value of uninhabited type Never at .<enum-variant(Ok)>.0.1
94	   |
95	   = note: The rules on what exactly is undefined behavior aren't clear, so this check might be overzealous. Please open an issue on the rustc repository if you believe it should not be considered undefined behavior.
96	

98	  --> $DIR/ub-enum.rs:92:1
99	   |
100	LL | const BAD_UNINHABITED_WITH_DATA2: Result<(i32, !), (i32, Never)> = unsafe { mem::transmute(1u64) };
-	   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed: encountered a value of uninhabited type Never at .<enum-variant(Err)>.0.1
+	   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed: encountered a value of the never type `!` at .<enum-variant(Ok)>.0.1
102	   |
103	   = note: The rules on what exactly is undefined behavior aren't clear, so this check might be overzealous. Please open an issue on the rustc repository if you believe it should not be considered undefined behavior.
104	
~~~~

~~~~
---- [ui] ui/suggestions/missing-trait-bounds-for-method-call.rs stdout ----
[ .. omitted for brevity .. ]
------------------------------------------
diff of stderr:

14	           which is required by `Foo<T>: Bar`
15	help: consider restricting the type parameters to satisfy the trait bounds
16	   |
-	LL | struct Foo<T> where T: Bar, T: std::default::Default {
+	LL | struct Foo<T> where T: std::default::Default, T: Bar {
18	   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
19	
20	error[E0599]: no method named `foo` found for reference `&Fin<T>` in the current scope
~~~~

CC @estebank and @RalfJung who wrote the tests, but I guess you may not be responsible for the underlying behavior.

--- a/src/librustc_typeck/check/method/suggest.rs
+++ b/src/librustc_typeck/check/method/suggest.rs
@@ -678,6 +678,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         .collect::<Vec<(usize, String)>>();
                     for ((span, empty_where), obligations) in type_params.into_iter() {
                         restrict_type_params = true;
+                        // #74886: Sort here so that the output is always the same.
+                        let mut obligations = obligations.into_iter().collect::<Vec<_>>();
+                        obligations.sort();
                         err.span_suggestion_verbose(
                             span,
                             &format!(
@@ -688,7 +691,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             format!(
                                 "{} {}",
                                 if empty_where { " where" } else { "," },
-                                obligations.into_iter().collect::<Vec<_>>().join(", ")
+                                obligations.join(", ")
                             ),
                             Applicability::MaybeIncorrect,
                         );

